challenges/02 It's super effective/app.js

```diff
@@ -1 +1,157 @@
+    ground: {fire: 2, electric: 2, grass: 0.5, poison: 2, flying: 0, bug: 0.5, rock: 2, steel: 2},
+    flying: {electric: 0.5, grass: 2, fighting: 2, bug: 2, rock: 0.5, steel: 0.5},
+    psychic: {fighting: 2, poison: 2, psychic: 0.5, dark: 0, steel: 0.5},
+    bug: {fire: 0.5, grass: 2, fighting: 0.5, poison: 0.5, fly: 0.5, psychic: 2, ghost: 0.5, dark: 2, steel: 0.5, fairy: 0.5},
+    rock: {fire: 2, ice: 2, fighting: 0.5, ground: 0.5, fly: 2, bug: 2, steel: 0.5},
+    ghost: {normal: 0, psychic: 2, ghost: 2, dark: 0.5},
+    dragon: {dragon: 2, steel: 0.5, fairy: 0},
+    dark: {fighting: 0.5, psychic: 2, ghost: 2, dark: 0.5, fairy: 0.5},
+    steel: {fire: 0.5, water: 0.5, electric: 0.5, ice: 2, rock: 2, steel: 0.5, fairy: 2},
+    fairy: {fire: 0.5, fighting: 0.5, poison: 0.5, ground: 2, dragon: 2, steel: 0.5}
+}
+
+/**Speichert alle Requests ab, die später nötig sind. */
+var requestsToDo = [undefined, undefined];
+
+/**
+ * Stellt eine Frage im Terminal und erwartet eine Antwort.
+ * Auf Basis der Antwort wird der Vergleich der Typen vorgenommen und
+ * wenn nötig, ein Api-Aufruf um Attacken-und Pokemon-Objekte abzurufen.
+ */
+rl.question('Lass den Vergleich beginnen: ', (answer) => {
+    let actors = splitAnswer(answer);
+    checkComposition(actors);
+    axios.all(requestsToDo)
+        .then((results) => {
+            if(results[0] != undefined) {
+                actors[0] = results[0].data.type.name;
+            }
+            if(results[1] != undefined) {
+                let pokemonTypes = [];
+                results[1].data.types.forEach(type => {
+                    pokemonTypes.push(type.type.name)
+                })
+                actors[1] = pokemonTypes
+            }
+            
+            let effectivityValue = effectivityCalculation(actors);
+            console.log('Der Effekt ist: ' + giveAnswerFromEffectivityValue(effectivityValue));
+            rl.close();
+        })
+})
+
+/**
+ * Unterteilt den eingegebenen String in Angreifer und Verteidiger.
+ *
+ * @param answer {string} - Benutzereingabe mit Angreifer und Verteidiger.
+ * @returns {Array|*} - Enthält auf: Index 0 -> Angreifer.
+ *                                   Index 1 -> Array aus Verteidigern.
+ */
+function splitAnswer(answer) {
+    let attackerAndDefender = answer.split(' -> ');
+    attackerAndDefender[1] = attackerAndDefender[1].split(' ');
+    return attackerAndDefender
+}
+
+/**
+ * Überprüft, welche Zusammensetzung aus Typen, Attacke und Pokemon übergeben wurde
+ * und fügt für eine Attacke oder ein Pokemon einen Api-Request in das requestsToDo-Array
+ * hinzu.
+ * 
+ * @param actors {Array} - Beinhaltet Typen, Attacke und Pokemon.
+ */
+function checkComposition(actors) {
+    if(!typeCharactaristics[actors[0]]){
+        requestsToDo[0] = getTypeFromAttack(actors[0])
+    }
+    if(!typeCharactaristics[actors[1][0]]) {
+        requestsToDo[1] = getTypeFromPokemon(actors[1])

```

Comment:
Ich bin mir nicht sicher, ob dir das bewusst war, aber vorsichtshalber merke ich das an: `actors[1]` ist an dieser Stelle immer ein Array. Ich bin mir nicht sicher, ob es mittlerweile Pokemon mit Leerzeichen im Namen gibt, aber so ein Name wäre hier in zwei Teile geteilt. Warum das ganze hier aber funktioniert (vorausgesetzt Pokemon haben keine Leerzeichen im Namen), liegt daran, dass `'poke' + ['mon'] == 'pokemon'`, aber `'poke + ['m', 'o', 'n'] == 'pokem,o,n'`.
Für das Verständnis besser das ganze als Kommentar vermerken oder explizit `actors[1]` in einen String umwandeln.